plugin.py

```diff
@@ -208,6 +208,30 @@ def on_pre_server_command(
             return True
         return False
 
+    def on_workspace_configuration(self, params: Dict, configuration: Any) -> Any:
+        if "section" not in params or "scopeUri" not in params:
+            return
+
+        session = self.weaksession()
+        if not session:
+            return
+
+        view = None
+        for view_protocol in session.session_views_async():
+            if view_protocol.get_uri() == params["scopeUri"]:
+                view = view_protocol.view
+                break
+
+        if not view:
+            return
+
+        if params["section"] == "java.format.tabSize":
+            return view.settings().get("tab_size", 4)
+        elif params["section"] == "java.format.insertSpaces":
+            return view.settings().get("translate_tabs_to_spaces")
+        else:
+            pass
```

Comment:
I don't know how server works exactly but don't you mean to fall back to returning `configuration` for any other request than the ones that need special treatment? Otherwise the server will not receive any of the settings from the `settings` objects` when it asks for `java.codeGeneration` section, for example?

Reply:
Oh yes, I missed that branch when I reworked the api.

Reply:
Fixed it.